test/AuthSamples.FunctionalTests/CustomPolicyProviderTests.cs

```diff
@@ -52,5 +119,26 @@ public CustomPolicyProviderTests(WebApplicationFactory<CustomPolicyProvider.Star
             Assert.Equal(HttpStatusCode.Redirect, response.StatusCode);
             Assert.Equal("http://localhost/account/signin?ReturnUrl=%2FHome%2FMinimumAge50", response.Headers.Location.ToString());
         }
+
+        internal static async Task<string> SignIn(HttpClient client, string userName, string dob)
+        {
+            var goToSignIn = await client.GetAsync("/account/signin");
+            var signIn = await TestAssert.IsHtmlDocumentAsync(goToSignIn);
+
+            var form = TestAssert.HasForm(signIn);
+            var registered = await client.SendAsync(form, new Dictionary<string, string>()
+            {
+                ["UserName"] = userName,
+                ["DOB"] = dob,
+            });
+
+            var registeredLocation = TestAssert.IsRedirect(registered);
+            if (registered.Headers.Contains("Set-Cookie"))
+            {
+                return registered.Headers.GetValues("Set-Cookie").FirstOrDefault();
```

Comment:
You can't echo the set-cookie header as the cookie header, their formats are different. You either have to parse it or ues @javiercn's idea for having the client do it for you.

Reply:
This is how all our functional tests do this I believe, I copied this from existing code that does our authentication tests...  and the test passes so...??

Reply:
But yeah I'm going to switch to the automatic client, that will be much nicer

Reply:
That's fishy. You may get lucky with some cookie values, but most will cause the Cookie header to be malformed. See https://github.com/aspnet/Security/blob/d95109c96d6b92d04178dea004583796824072da/test/Microsoft.AspNetCore.Authentication.Test/WsFederation/WsFederationTest.cs#L255-L265

Reply:
Yeah well its been that way in our Security tests for ages... so maybe that needs to be fixed?  But we use it everywhere

https://github.com/aspnet/Security/blob/dev/test/Microsoft.AspNetCore.Authentication.Test/TestExtensions.cs#L36

Reply:
@Tratcher Can you follow up to see if there's anything we need to clean up in our TestExtensions?  I'm going to switch to using the new API for auth samples so it won't be an issue for this repo

Reply:
Ah, this copy does it better: https://github.com/aspnet/Security/blob/ff9f145a8e89c9756ea12ff10c6d47f2f7eb345f/test/Microsoft.Owin.Security.Interop.Test/CookieInteropTests.cs#L305

It trims off all the properties and stores it in another field that is safe to echo. It only works for one cookie, but it's a start.

Reply:
FYI I don't see any of the auth tests using SetCookie incorrectly. Most of them examine the contents of SetCookie. A few of them correctly parse SetCookie before trying to echo it. I don't see any that echo it in its raw form.